examples/energy-management-app/energy-management-common/src/EnergyEvseDelegateImpl.cpp

```diff
@@ -1492,6 +1690,55 @@ CHIP_ERROR GetEpochTS(uint32_t & chipEpoch)
     return CHIP_NO_ERROR;
 }
 
+/**
+ * @brief   Helper function to get current timestamp and work out the day of week
+ *
+ * NOTE that the time_t is converted using localtime to provide the timestamp
+ * in local time. If this is not supported on some platforms an alternative
+ * implementation may be required.
+ *
+ * @param   unixEpoch (as time_t)
+ *
+ * @return  bitmap value for day of week
+ * Sunday = 0x01, Monday = 0x01 ... Saturday = 0x40 (1<<6)
+ */
+uint8_t GetDayOfWeekUnixEpoch(time_t unixEpoch)
+{
+    // Define a timezone structure and initialize it to the local timezone
+    // This will capture any daylight saving time changes
+    struct tm local_time;
```

Comment:
This doesn't work across platforms, does it? Does this cluster tie into the time sync cluster? It should be getting local time from there. Many smaller impls won't be carrying their own dst tables.

Reply:
Agreed - the comment in the function description also says that this would only work on platforms that do this. Again this sits in the examples area (so manufacturers can come up with their own implementations if they wanted to adapt it).

 We deliberately have mandated use of TimeSync (e.g. you could use NTP or a real-time clock or some other way of getting real time) but TimeSync is definitely an option people could use.

Reply:
Marking as resolved.